docs/faq.rst

```diff
@@ -97,29 +96,27 @@ Computations involving uncommitted data are performed on the default
 device and the results are uncommitted on the default device.
 
 Data can also be placed explicitly on a device using :func:`jax.device_put`
-with a ``device`` parameter, in which case if becomes **committed** to the device:
+with a ``device`` parameter, in which case it—the data—becomes **committed** to the device:
 
 >>> import jax
 >>> from jax import device_put
 >>> print(device_put(1, jax.devices()[2]).device_buffer.device())  # doctest: +SKIP
 gpu:2
 
-Computations involving some committed inputs, will happen on the
-committed device, and the result will be committed on the
-same device. It is an error to invoke an operation on
-arguments that are committed to more than one device.
+Computations involving some committed inputs will happen on the
+committed device and the result will be committed on the
+same device. You should not invoke an operation on
+arguments that are committed to more than one device—it will lead to errors.
```

Comment:
Maybe:
Invoking an operation on arguments that are committed to more than one device will raise an error.

WDYT? "It will lead to errors" implies to me that something will eventually go wrong down the line, but I believe this should immediately raise a hopefully-clear error.

Reply:
I agree ⚡︎